src/message_filters/__init__.py

```diff
@@ -76,12 +78,34 @@ class Subscriber(SimpleFilter):
     to it.
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(
+        self,
+        node: Node,
+        msg_type: Type[MsgT],
+        topic: str,
+        qos_profile: Union[QoSProfile, int] = 10,
+        **kwargs,
+    ):
+        """
+        Construct a Subscriber.
+
+        Args:
+        ----
+        node (Node): The node to create a subscriber for.
+        msg_type (Type[MsgT]): The type of ROS messages the subscription will subscribe to.
+        topic (str): The name of the topic the subscription will subscribe to.
+        qos_profile (QoSProfile | int): A QoSProfile or a history depth to apply to the
+            subscription. In the case that a history depth is provided, the QoS history is
+            set to KEEP_LAST, the QoS history depth is set to the value of the parameter,
+            and all other QoS settings are set to their default values.
+
+        """
         SimpleFilter.__init__(self)
-        self.node = args[0]
-        self.topic = args[2]
-        kwargs.setdefault('qos_profile', 10)
-        self.sub = self.node.create_subscription(*args[1:], self.callback, **kwargs)
+        self.node = node
+        self.topic = topic
+        self.sub = self.node.create_subscription(
```

Comment:
```suggestion
        self.subscription = self.node.create_subscription(
```
?

Do not insist. Just asking.

LGTM otherwise. @mjcarroll ?

Reply:
That could break downstream users if they access the `sub` attribute, so I think we should keep `sub`.

Reply:
Makes sense.

Not being completely serious: Maybe add a property for backwards compatibility? :grin: